Software/Development/SemesterProject/App/dutyCycle.c

```diff
@@ -0,0 +1,192 @@
+{
+    DutyCycle legA;
+    DutyCycle legB;
+    DutyCycle legC;
+}dutyCycle;
+
+
+/* CONSIDER REVISION to reduce computing load.
+ * Function for internal use.
+ * map function performs a linear regression.
+ * Inputs are a value with its corresponding lower and upper limits
+ * as well as the lower and upper limits of the variable to return.
+ */
+Uint16 mapDuty(float x, float x_min, float x_max, Uint16 y_min, Uint16 y_max)
+{
+    return y_min + (Uint16)((float)(y_max-y_min)*(x-x_min)/(x_max-x_min));
+}
+
+/*
+ * Function for internal use.
+ * The duty cycle is checked in case it lies outside of boundaries.
+ * Later the struct containing the duty information gets updated.
+ */
+void fillDutyStruct(float duty, int leg)
+{
+
+    //Counter compare module uses the time counter variable to compare in order to set the
+    //correct PWM duty cycle, for this, it is important to know how many counts there are in a cycle.
+
+    //In the PWMConfig.c file, the period of the duty cycle is set by assigning the number of clk of a
+    //a period, this is done in the command 'EPwm1Regs.TBPRD = INTERNAL_FREQ/(2*SW_FREQ);'
+    //Since the configured carrier is a symmetrical triangular, the real period is actually twice as
+    //much but the counter will reach maximum that value.
+
+    //duty cycle float variable must then be converted into a Uint16 with correspondent
+    //input 0.00 -> output 0 and input 100.00 -> output INTERNAL_FREQ/(2*SW_FREQ).
+
+    //Error check.
+    if (duty > 100)
+    {
+        //Act accordingly.
+        //REPORT ERROR.
+        duty = 100; //For now, adjust to limit.
+    }
+    else if (duty < 0)
+    {
+        //Act accordingly.
+        //REPORT ERROR
+        duty = 0; //For now, adjust to limit.
+    }
+
+
+    //No errors so save to struct.
+    switch (leg)
+    {
+        case LEG_A:
+        {
+
+            dutyCycle.legA.dutyValue = duty;
+
+            //Transform duty to comparator.
+            dutyCycle.legA.dutyCompare = mapDuty(duty, 0.0, 100.0, 0, INTERNAL_FREQ/(2*SW_FREQ));
+        }break;
+
+        case LEG_B:
+        {
+            dutyCycle.legB.dutyValue = duty;
+
+            //Transform duty to comparator.
+            dutyCycle.legB.dutyCompare = mapDuty(duty, 0.0, 100.0, 0, INTERNAL_FREQ/(2*SW_FREQ));
+        }break;
+        case LEG_C:
+        {
+            dutyCycle.legC.dutyValue = duty;
+
+            //Transform duty to comparator.
+            dutyCycle.legC.dutyCompare = mapDuty(duty, 0.0, 100.0, 0, INTERNAL_FREQ/(2*SW_FREQ));
+        }break;
+
+        default:
+        {
+
+        }break;
+    }
+
+}
+
+
+/*
+ * Duty cycle is set on the leg A.
+ * This function expects an input duty value that is the corresponding PWM duty to be set.
+ * duty cycle is a float ranging from 0 to 100.
+ */
+void setDutyA(float duty)
+{
+    fillDutyStruct(duty, LEG_A);
+
+    //Set counter compare to correct value.
+    EPwm1Regs.CMPA.half.CMPA = dutyCycle.legA.dutyCompare;
```

Comment:
Here again the casual developer coming around will have no clue what is going on. A way of clarifying stuff would be to wrap this function inside a higher abstraction level function, the same abstraction level as filldutyStruct()

same for A, B and C legs

Reply:
Are you suggesting to create 3 new functions which have only this line inside and that are only called once in the code?
I see your point but maybe it's a bit too much and it is something that could be solved with a nice comment... Idk, what do you guys think?